src/nvim/screen.c

```diff
@@ -3052,9 +3052,11 @@ win_line (
           diff_hlf = HLF_CHD;                   // changed line
         }
         line_attr = win_hl_attr(wp, diff_hlf);
-        // Overlay CursorLine onto diff highlight, unless it's low-priority.
-        if (!line_attr_lowprio && wp->w_p_cul && lnum == wp->w_cursor.lnum) {
-          line_attr = hl_combine_attr(line_attr, win_hl_attr(wp, HLF_CUL));
+        // Overlay CursorLine onto diff-mode highlight.
+        if (wp->w_p_cul && lnum == wp->w_cursor.lnum) {
+          line_attr = 0 != line_attr_lowprio  // Low-priority CursorLine
+            ? hl_combine_attr(line_attr, hl_get_underline())
```

Comment:
Why not `hl_combine_attr(win_hl_attr(wp, HLF_CUL), line_attr)` which would work for any attribute and not only `underline` ?

Reply:
@bfredl Thanks. That effectively hides the CursorLine with the use-case given above, but maybe we could do that + add underline. The idea is to have something that's visible without being unreadable.

Reply:
@justinmk How could `hl_combine_attr(win_hl_attr(wp, HLF_CUL), line_attr)`  "effectively hide underline" in any way that `hl_combine_attr(line_attr, hl_get_underline())` doesn't. The former will include _any_ attribute that CursorLine contains.

Reply:
If the purpose rather is to _actively_ special case CursorLine on diff lines, then `CursorLineDiff` group would be the proper and configurable option (which even can default to underline, for this behavior by default) , rather than adding _ad-hoc_ logic for hardcoding an implicit `gui=underline` group, which is not done anywhere else. What is the motivation to add code to _prohibit_ users to configure this particular case?

Reply:
I mean visually, the diff highlight still dominates. Try the command given in the top description.

Reply:
Again, that depends on the attributes used. If `reverse` is used, both high and low priority already works, but with this change diff lines are needlessly different.

Reply:
Yes, it depends on the attributes.

The point of this PR is to avoid a degenerate case for something like `hi CursorLine ctermbg=grey`. With `hi DiffAdd ctermfg=black ctermbg=red`, that's hard to read.

Reply:
Yes, all I'm saying is if the corner case is a result of user configuration,  the override itself should be configurable (compare CursorLineNr). Otherwise it will just make some configurations better by making others worse..

Reply:
> But with the latest push AFAIK it won't make anything worse except if someone considers it worse to have an underline. It's also overridable by setting foreground on CursorLine (so it's not "low priority").

@justinmk All other screen attributes added by vim/nvim are configurable by the user, moving towards a mixture of user configurable attributes and hard-coded attributes is a step backwards, and an unexpected one.  If it counts for anything, it also breaks my setup (i don't use `CursorLine` always, but when I do), by not only unexpected underline but also unreadable text on diff lines. And yes it is "fixable" by setting guifg, but why should I or any user need to use guifg if we only want guibg (or some non-ul attribute)? Why should anyone be forced to use guifg to take control of bg and attributes, instead of having the later being controlled by `guibg=` and `gui=` like in all other UI highlights in nvim?